lib/machete/matchers/iso8601_datetime_matcher.ex

```diff
@@ -85,10 +89,15 @@ defmodule Machete.ISO8601DateTimeMatcher do
 
   defimpl Machete.Matchable do
     def mismatches(%@for{} = a, b) when is_binary(b) do
-      DateTime.from_iso8601(b)
-      |> case do
-        {:ok, datetime_b, 0} -> datetime_b ~>> datetime(a.datetime_opts)
-        _ -> mismatch("#{inspect(b)} is not a parseable ISO8601 datetime")
+      case DateTime.from_iso8601(b) do
+        {:ok, datetime_b, 0} ->
+          datetime_b ~>> datetime(a.datetime_opts)
+
+        _ ->
+          case NaiveDateTime.from_iso8601(b) do
+            {:ok, naive_datetime_b} -> naive_datetime_b ~>> naive_datetime(a.datetime_opts)
+            _ -> mismatch("#{inspect(b)} is not a parseable ISO8601 datetime")
+          end
```

Comment:
This ends up possibly matching naive 8601s when you otherwise may not have meant to (TBH, I'd never realized 8601s were valid *without* a timezone until you brought it up!). I agree that this belongs in this matcher, but maybe hidden behind a flag? 8601 calls that last component an 'offset', so maybe something like `offset_required`, defaulting to true?

Reply:
Yeah that makes sense, I've now added it as an option :+1: